monai/transforms/spatial/dictionary.py

```diff
@@ -1285,7 +1285,7 @@ def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.nda
         d = dict(data)
         if not self._do_transform:
             for key in self.keys:
-                self.push_transform(d, key, extra_info={"rot_mat": np.eye(4)})
+                self.push_transform(d, key, extra_info={"rot_mat": np.eye(d[key].ndim)})
```

Comment:
Should this be ndim+1? Not sure if it's purely rotation matrix or also translation component.

Reply:
Sorry, thought about this a bit more, and it makes sense.

Reply:
yes it's with the channel dimension...